Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/Vehicle.hs

```diff
@@ -124,9 +124,12 @@ findAllByVariantRegNumMerchantId variantM mbRegNum limitVal offsetVal (Id mercha
                   $ B.all_ (BeamCommon.vehicle BeamCommon.atlasDB)
   catMaybes <$> mapM fromTType' (fromRight [] vehicles)
 
-findByRegistrationNo :: (MonadFlow m) => Text -> m (Maybe Vehicle)
+findByRegistrationNo :: MonadFlow m => Text -> m (Maybe Vehicle)
 findByRegistrationNo registrationNo = findOneWithKV [Se.Is BeamV.registrationNo $ Se.Eq registrationNo]
 
+findAllByFleetOwnerId :: MonadFlow m => Text -> m [Vehicle]
+findAllByFleetOwnerId fleetOwnerId = findAllWithKV [Se.Is BeamV.fleetOwnerId $ Se.Eq $ Just fleetOwnerId]
```

Comment:
It's better to have a limit else if number of vehicle of a fleet owner increase this API will become very slow.

Reply:
yes , but if we add limit here then the total stats will not be proper. like we can have limit  in the number of vehicle that can be a part of fleet.